UnitTests/MockCaptureFixture.cs

```diff
@@ -0,0 +1,163 @@
+            mock.Setup(x => x.DoSomething(capture.CaptureAny()));
+
+            mock.Object.DoSomething("Hello!");
+            
+            var expectedValues = new List<string> { "Hello!" };
+            Assert.Equal(expectedValues, capture.All);
+        }
+
+        [Fact]
+        public void CanCaptureMultipleParameters()
+        {
+            var capture = new MockCapture<string>();
+            var mock = new Mock<IFoo>();
+            mock.Setup(x => x.DoSomething(capture.CaptureAny()));
+
+            mock.Object.DoSomething("Hello!");
+            mock.Object.DoSomething("World!");
+
+            var expectedValues = new List<string> { "Hello!", "World!" };
+            Assert.Equal(expectedValues, capture.All);
+        }
+
+        [Fact]
+        public void CanCaptureSpecificParameter()
+        {
+            var capture = new MockCapture<string>();
+            var mock = new Mock<IFoo>();
+            mock.Setup(x => x.DoSomething(capture.Capture(p => p.StartsWith("W"))));
+
+            mock.Object.DoSomething("Hello!");
+            mock.Object.DoSomething("World!");
+
+            var expectedValues = new List<string> { "World!" };
+            Assert.Equal(expectedValues, capture.All);
+        }
+
+        [Fact]
+        public void CanCaptureLastParameter()
+        {
+            var capture = new MockCapture<string>();
+            var mock = new Mock<IFoo>();
+            mock.Setup(x => x.DoSomething(capture.CaptureAny()));
+
+            mock.Object.DoSomething("Hello!");
+            mock.Object.DoSomething("World!");
+
+            Assert.Equal("World!", capture.Last);
+        }
+
+        [Fact]
+        public void CanCaptureSingleParameter()
+        {
+            var capture = new MockCapture<string>();
+            var mock = new Mock<IFoo>();
+            mock.Setup(x => x.DoSomething(capture.CaptureAny()));
+
+            mock.Object.DoSomething("Hello!");
+
+            Assert.Equal("Hello!", capture.Single);
+        }
+
+        [Fact]
+        public void ThrowsIfSingleIsCalledWithNoParameter()
+        {
+            var capture = new MockCapture<string>();
+            var mock = new Mock<IFoo>();
+            mock.Setup(x => x.DoSomething(capture.CaptureAny()));
+
+            Assert.Throws<InvalidOperationException>(() => capture.Single);
+        }
+
+        [Fact]
+        public void ThrowsIfSingleIsCalledWithManyParameter()
+        {
+            var capture = new MockCapture<string>();
+            var mock = new Mock<IFoo>();
+            mock.Setup(x => x.DoSomething(capture.CaptureAny()));
+
+            mock.Object.DoSomething("Hello!");
+            mock.Object.DoSomething("World!");
+
+            Assert.Throws<InvalidOperationException>(() => capture.Single);
+        }
+
+        [Fact]
+        public void CanRunCallbackOnParameter()
+        {
+            var capture = new MockCapture<FooParam>();
+            capture.RunOnCapture(x => x.Value = 42);
```

Comment:
This seems to be somewhat overlapping with a `Callback` before returns, which would also allow to mutate the value before returning it. 

The test even uses it only for that, never actually using the captured parameter. It's almost like a callback based on the parameter type, rather than a particular setup. Not sure about mixing that behavior with the rest.

Maybe make the MockCapture could have virtual `Any` implementations that could be further extended to add that if needed for your particular usage? I almost feel the same way about the copier stuff...